tests/libvmi/factory.go

```diff
@@ -43,48 +43,55 @@ func NewFedora(opts ...Option) *kvirtv1.VirtualMachineInstance {
 		WithRng(),
 		WithContainerImage(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling)),
 	}
-	opts = append(fedoraOptions, opts...)
+	opts = append(opts, fedoraOptions...)
```

Comment:
I'm unsure why you swapped them, but I think this is problematic.
The inputted options should have the chance to override any defaults, but now it is not the case.

Reply:
You are totally right, and the last commit has the right order. 
It looks like I need to cleanup/split the commits